atlasdb-dbkvs-tests/src/test/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/oracle/OracleAlterTableIntegrationTest.java

```diff
@@ -101,6 +105,25 @@ public void after() {
         connectionSupplier.close();
     }
 
+    @Test
+    public void whenConfiguredAlterMetadataToMatchTableSchemaAndOldDataIsStillReadable() {
+        defaultKvs.createTable(TABLE_REFERENCE, EXPECTED_TABLE_METADATA.persistToBytes());
+        writeData(defaultKvs, ROW_1, TIMESTAMP_1);
+        assertThatDataCanBeRead(defaultKvs, DEFAULT_CELL_1, TIMESTAMP_1, DEFAULT_VALUE_1);
+
+        modifyMetadataToNotHaveOverflow();
+        // metadata is cached, thus we must create a new kvs to forcefully refresh the cache
+        KeyValueService badKvs = createKvs(DbKvsOracleTestSuite.getKvsConfig());
+        assertThatThrownBy(() -> writeData(badKvs, ROW_2, TIMESTAMP_2));
```

Comment:
For context: this was the line I described as "this looks sus" in the first review, but admittedly I should have gone into more detail :)

The way this test is written in won't really work, because a KVS that believes no overflow column is used _can_ write to a store in which it is used - the throwing behaviour only happens when a KVS that believes an overflow column is used writes to one in which it is not. 

The `badKvs` thinks that the table is a raw data table: this means that it will perform writes by doing a statement of the form `INSERT INTO foo (row_name, col_name, ts, value) VALUES (1, 2, 3, 4)`... **but this is a perfectly valid statement to insert into an overflow table**, which has five columns `(row_name, col_name, ts, value, overflow)` where neither `value` nor `overflow` are required to be non-null (since in practice one of them only should be present, or we have Local-Remote Corruption (TM)).

So I would expect this test to fail on the basis of `writeData(...)` not actually throwing an exception, and checking circle looks like that is indeed the case.

Reply:
Yup ok we're on the same page! This PR shouldn't be needed then, as the metadata indicates that it should have a `RAW` overflow value, but the table schema has an overflow column, which isn't used/needed. Thus, I think we can close this out!